### PR TITLE
sending mail to users when new user from same college comes

### PR DESCRIPTION
**Pull request by gauthamzz**

### CODE REVIEW

File: models/db.py [hunk 1]
```diff
@@ -184,7 +184,29 @@ def sanitize_fields(form):
...
+    rows = db(query).select(atable.email, atable.stopstalk_handle)
+    subject = "New user registered from your college"
+    message = """
+Hello %s,
+%s from your college has just joined StopStalk
+Send a friend request now %s for better experience on StopStalk
+Regards,
+StopStalk
+              """ % (row.stopstalk_handle,
+                        form.vars.first_name + " " + form.vars.last_name,
```
Comment (raj454raj) [inline models/db.py]: Match the arguments to the last starting parenthesis.
Example:
```
f(arg1,
  arg2,
  arg3(arg4,
       arg5))
```

File: models/db.py [hunk 2]
```diff
@@ -184,7 +184,29 @@ def sanitize_fields(form):
     for site in current.SITES:
         site_handle = site.lower() + "_handle"
         form.vars[site_handle] = form.vars[site_handle].lower()
-
+#-----------------------------------------------------------------------------
+def inside_institute(form):
```
Comment (raj454raj) [inline models/db.py]: `notify_institute_users` would be more informative.

File: models/db.py (hunk 1 above)
Comment (gauthamzz) [inline models/db.py]: i think its correct

File: models/db.py [hunk 3]
```diff
@@ -197,13 +197,10 @@ def inside_institute(form):
...
 StopStalk
               """ % (row.stopstalk_handle,
-                        form.vars.first_name + " " + form.vars.last_name,
-                        URL('default', 'search'),
-                     URL("default", 
-                             "unsubscribe", 
-                             scheme=True, 
-                             host=True, 
-                             extension=False))    
+                    form.vars.first_name + " " + form.vars.last_name,
```
Comment (raj454raj) [inline models/db.py]: One more space. Refer other parts of the code.

Comment (gauthamzz) [inline models/db.py]: I changed it , view the updated one

File: models/db.py [hunk 4]
```diff
@@ -184,7 +184,26 @@ def sanitize_fields(form):
     for site in current.SITES:
         site_handle = site.lower() + "_handle"
         form.vars[site_handle] = form.vars[site_handle].lower()
-
+#-----------------------------------------------------------------------------
+def notify_institute_users(form):
+    # Send mail to all users form the same college when someone registers
```
Comment (raj454raj) [inline models/db.py]: Typo: *from

**Comment by raj454raj** [comment]
@gauthamzz : Still that one space after every argument is left. Have a look at similar indentations in the code.

File: models/db.py [hunk 5]
```diff
@@ -184,7 +184,37 @@ def sanitize_fields(form):
     for site in current.SITES:
         site_handle = site.lower() + "_handle"
         form.vars[site_handle] = form.vars[site_handle].lower()
-
+#-----------------------------------------------------------------------------
```
Comment (raj454raj) [inline models/db.py]: One more newline between functions.

File: models/db.py [hunk 6]
```diff
@@ -184,7 +184,26 @@ def sanitize_fields(form):
...
+    subject = "New user registered from your college"
+    message = """
+Hello %s,
+%s from your college has just joined StopStalk
+Send a friend request now %s for better experience on StopStalk
+Regards,
+StopStalk
+              """ % (row.stopstalk_handle,
+                    form.vars.first_name + " " + form.vars.last_name,
+                    URL('default', 'search'),
```
Comment (raj454raj) [inline models/db.py]: Double quotes for Python

File: models/db.py [hunk 7]
```diff
@@ -184,7 +184,26 @@ def sanitize_fields(form):
...
+                    URL("default", "unsubscribe", scheme=True, host=True, extension=False)
+                    )    
+        for row in rows:
```
Comment (raj454raj) [inline models/db.py]: Unexpected indent error here.

File: models/db.py [hunk 8]
```diff
@@ -184,7 +184,26 @@ def sanitize_fields(form):
     for site in current.SITES:
         site_handle = site.lower() + "_handle"
         form.vars[site_handle] = form.vars[site_handle].lower()
-
+#-----------------------------------------------------------------------------
+def notify_institute_users(form):
+    # Send mail to all users from the same institute when someone registers
+    query = (atable.institute == form.vars.institute)
```
Comment (raj454raj) [inline models/db.py]: atable undefined.
```
atable = db.auth_user
```

File: models/db.py [hunk 9]
```diff
@@ -184,7 +184,26 @@ def sanitize_fields(form):
...
+    rows = db(query).select(atable.email, atable.stopstalk_handle)
+    subject = "New user registered from your college"
+    message = """
+Hello %s,
+%s from your college has just joined StopStalk
+Send a friend request now %s for better experience on StopStalk
+Regards,
+StopStalk
+              """ % (row.stopstalk_handle,
```
Comment (raj454raj) [inline models/db.py]: row undefined here.
Put this message assignment in the next for loop

File: models/db.py [hunk 10]
```diff
@@ -184,7 +184,26 @@ def sanitize_fields(form):
...
+def notify_institute_users(form):
+    # Send mail to all users from the same institute when someone registers
+    query = (atable.institute == form.vars.institute)
+    rows = db(query).select(atable.email, atable.stopstalk_handle)
+    subject = "New user registered from your college"
+    message = """
+Hello %s,
+%s from your college has just joined StopStalk
+Send a friend request now %s for better experience on StopStalk
+Regards,
```
Comment (raj454raj) [inline models/db.py]: You forgot to add - `To stop receiving mails - %s`  here

File: models/db.py (hunk 6 above)
Comment (raj454raj) [inline models/db.py]: Also add the scheme=True, host=True and extension=False arguments here.

**Comment by raj454raj** [comment]
@gauthamzz : I have solved the issue. Just for your reference - https://github.com/stopstalk/stopstalk-deployment/commit/491c16f06a6a9010bef2d45deb1822d3f726673d
Hope to see you solve other similar issues !

**Pull request closed without merging**